sdk/daml-script/test/daml/upgrades/ContractKeys.daml

```diff
@@ -16,14 +21,44 @@ versions: 2
 
 main : TestTree
 main = tests
-  [ ("Query an unchanged old key for a new contract", queryKeyUnchanged)
-  , ("ExerciseByKey command an unchanged old key for a new contract", exerciseCmdKeyUnchanged)
-  , ("Fetching an unchanged old key for a new contract", fetchKeyUnchanged)
-  , ("ExerciseByKey in Update an unchanged old key for a new contract", exerciseUpdateKeyUnchanged)
-  , ("Query an Upgraded old key for a new contract", queryKeyUpgraded)
-  , ("ExerciseByKey command an Upgraded old key for a new contract", exerciseCmdKeyUpgraded)
-  , ("Fetching an Upgraded old key for a new contract", fetchKeyUpgraded)
-  , ("ExerciseByKey in Update an Upgraded old key for a new contract", exerciseUpdateKeyUpgraded)
+  [ (,) "Unchanged key type" $ tests
```

Comment:
These were getting quite repetitive so I decided to make the `TestTree` actually a tree, let me know what you think @samuel-williams-da @paulbrauner-da

Reply:
Either way, looks nice, could consider an alias for `(,)` but thats nitpicking

Reply:
@samuel-williams-da it works in the sense that the nested test cases run and failure/success is propagated correctly, but the logging isn't great. I've opened https://github.com/digital-asset/daml/pull/19936 to improve that while adding the alias for `subtree`s